specs/logging_implementation.adoc

```diff
@@ -0,0 +1,201 @@
+= Generic message mechanism for logs and alerts in Tendrl"
+
+Logs and alerts need to be treated as messages emitted by Tendrl.
+
+== Problem description
+* The logs, alerts and flight_operations_updates in Tendrl are handled differently
+  and there is no standard mechanism which makes sure no internal details are
+  exposed as part of alerts/notifications sent to end user.
+* Ideally logs messages, flight operation updates and alerts are messages which
+  are emitted from an application. Tendrl needs to treat them as one entity and
+  a common handler should decide what to be done with the message which is emitted
+  and received at handler.
+== Use Cases
+
+Logs and alerts need to be treated as messages.
+
+== Proposed change
+* Logs and alerts need to be treated as messages emitted by Tendrl modules.
+* A schema needs to be defined for emitted messages.
+* Message priorities, types etc. need to be defined at the time of emission.
+* The integration/bridge applications must pass the message as per schema.
+* For messages that serve as notifications to the administrator (or alerts),
+  exact error message must be hardcoded by the developers. No backtraces or
+  system generated error messages should be sent as notifications to the
+  administrator.
+* Notifications sent to the administrator should provide sufficient context to
+  enable debugging of the problem via investigation into the log files.
+* Notifications and alerts need to be delivered to the node agent alert sockets.
+* Commons needs to implement a logging library that hides all the implementation
+  details and configuration options from any of the components using the library.
+  Components should be able to blindly emit messages to the Message Handler module
+  in Tendrl and it should be able to deliver them to either syslog, log files,
+  alert sockets etc. depending upon the metadata and configuration.
+* Syslog must be the default logging configuration. Logging to individual
+  configuration files should be thoroughly discouraged.
+
+== Alternatives
+
+* None
+
+== Data model impact
+
+None
+
+== Impacted modules
+
+=== Tendrl/common impact
+* Create a new class called "Message" in common for message handling with attributes
+  handler_name, message_type, message_id, log_level, message_tag, message, severity,
+  timestamp. (other attributes are filename, function_name, line_number, timestamp.
+  These details are appended with message automatically).
+```
+Message class object structure
+     log_handler_name
+     message_type(log/alert/flight_operation)
+     log_level(info/error..)
+     message
+     message_trace_back_flag(True/False)
+     message_uuid (optional for log)
```

Comment:
yes actually my logic also same why i mentioned here as optional means, for flight operation i am using job id as message id, for alerting i am using alert id as message id but what about normal exception logs. It does not required any id. What i try to mentioned here is for normal log messages it is optional